ext/ddtrace_profiling_native_extension/ruby_helpers.c

```diff
@@ -118,3 +118,39 @@ char* ruby_strndup(const char *str, size_t size) {
 
   return dup;
 }
+
+VALUE _id2ref(VALUE obj_id) {
+  // Call ::ObjectSpace._id2ref natively. It will raise if the id is no longer valid
+  VALUE module_object_space = rb_define_module("ObjectSpace");
```

Comment:
Minor: If we do end up calling `_id2ref` through `rb_funcall`, we can cache this somewhere